### ISSUE TITLE
Displaying integer-valued real numbers

### ISSUE THREAD

**Issue by mbillingr** [mbillingr/interpreter#6]
The interpreter prints real numbers like integers if their fractional part is zero.
Example:
```
>> 1.0
1
```

With small numbers this is no big deal but it may cause confusion if the fractional part is lost due to the limited precision of the floating point representation.
```
>> (define a 1234567890.12)
>> (* a a)
1524157875315348200
```

`1524157875315348200`  looks like an integer, when in fact it is a floating point number with limited precision. Printing the number as `1524157875315348200.0` would communicate that fact more clearly.
Thank you @pgl10 for reporting!

I can see several options to fix this:
- [x] (*easy*) Use the debug formatter, which prints floats with `.0` at the end but never uses scientific notation ([Example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a05acce11335a1a86e87a087a8ef7a9b)).
- [ ] (*easy*) Wait for the [`{:g?}` format switch](https://github.com/rust-lang/rfcs/pull/2729) which may never come.
- [ ] (*moderate*) Implement custom formatting to properly show the available precision.
- [ ] (*trivial*) Leave as it is, arguing that the user should not be concerned with the internal implementation of numbers. This argument breaks down for large numbers.

At the moment, I prefer the first option. The trailing zeros are a bit ugly for large numbers but it's so easy to implement.

Any thoughts?

**Comment by pgl10** [comment]
It is not so easy to display always a real number as a real. Sometimes it looks like an integer. I know a way to display always a real number as a real in C++ and I let you know about this, but I don't know Rust.

**Comment by mbillingr** [comment]
I think there is an easy way in Rust to it. Otherwise I'll try to adopt your C++ solution.

**Comment by mbillingr** [comment]
Implemented in f2a781fb3336636ce5291b903fea6ce7bca00acb
